docs/disclosure-log.md

```diff
@@ -0,0 +1,96 @@
+
+```
+    if (localEnd.state === INIT && remoteEnd === null)
+```
+
+The correct code should include both the cases when a single chain is in state `INIT`, as well as the case when both chains are in state `INIT`, as specified here: [Relayer.tla](https://github.com/informalsystems/ibc-rs/blob/master/docs/spec/relayer/Relayer.tla#L174)
+This fix only concerns the relayer algorithm ICS018.
+
+##### Channel handshake (ICS4) liveness problem
+
+The same issue (and fix) seems to exist for the channel handshake datagrams.
+
+
+### 2. ICS3 liveness problem due to `UpdateClient` semantics
+
+This problem is not specific to the connection handshake protocol (ICS3) itself, but is a bug in the way the relayers use the `UpdateClient` action.
+We classify this under ICS3, however, since this is the context where we discovered the problem.
+The TLA+ spec we have for depicting this liveness problem is for the ICS3 protocol.
+
+##### Problem statement
+
+Related issues: [#71](https://github.com/informalsystems/ibc-rs/issues/71) and [#61](https://github.com/informalsystems/ibc-rs/issues/61).
+The problem is more thorougly described in #61, but for completeness sake we restated it here in a compact form.
+
+The liveness property that a correct relayer should provide is eventual delivery.
+Assuming some source chain `A`, destination chain `B`, and an IBC item `X` (e.g., connection, channel, or packet) on chain `A`, we can define this property as follows:
+
+> For any IBC item `X` on chain `A` destined for chain `B`, eventually, a correct relayer will submit item `X` to chain `B`.
+
+This is difficult to guarantee in practice, however.
+Intuitively, the difficulty arises because of a combination of two factors:
+
+1. __Proof requirement:__ For chain `B` to accept item `X`, it must verify the authenticity of this item; this is done via the light client that chain `B` maintains.
+Given an item `X` and a commitment proof for `X` constructed at height `h-1`, the light client requires the consensus state at height `h` that includes that commitment root required for verification.
+
+2. __Concurrency:__ Different relayers may update the same light client.
+Suppose a relayer `r1` wants to submit a consensus state at height `h`.
+In the meantime, however, another relayer `r2` may update this same light client to height `h'`.
+Assume `h'` is bigger than `h`.
+Now `r1` will be unable to update the light client with consensus state `h`, because the light client disallows updates with smaller heights than the current height `h'`.
+Consequently, the relayer will be unable to submit consensus state at height `h`.
```

Comment:
```suggestion
Consequently, the relayer will be unable to submit message for `X`.
```

Reply:
`X` is not mentioned in (2/Concurrency), so it would be confusing to talk about it here.